Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -36,13 +44,50 @@ class OpenSearchEngine {
     }
 
     /**
+     * Return the arg that we use for searching for this engine
+     * Problem: the search terms may not be a query arg, they may be part of the URL - how to deal with this?
+     **/
+    private func getQueryArgFromTemplate() -> String? {
+        if let queryStartIndex = searchTemplate.rangeOfString("?")?.startIndex.successor() {
+            let queryArgs = searchTemplate.substringFromIndex(queryStartIndex)
+            let queryParts = queryArgs.componentsSeparatedByString("&").map { $0.componentsSeparatedByString("=") }
+            for part in queryParts {
+                if part[1] == SearchTermComponent {
+                    return part[0]
+                }
+            }
+
+        }
+        return nil
+    }
+
+    /**
+     * check that the URL host contains the name of the search engine somewhere inside it
+     **/
+    private func isSearchURLForEngine(url: NSURL?) -> Bool {
+        guard let urlHost = url?.host else { return false }
+        return urlHost.localizedCaseInsensitiveContainsString(self.shortName)
```

Comment:
Hm, still not very excited about this function. Assuming the URL will contain the search engine name is quite fragile, and I'm guessing at least one of our many localized search engines won't pass this test. (Yep, just verified: one arbitrary engine that won't work is `SearchPlugins/uz/wikipedia-uz.xml`, which has a short name of "Vikipediya (uz)").

Could we pull the entire host out of the template URL and see if it's the same host as the page in question?